### DISCUSSION TITLE
How work certificat? How force generate certificat?

### DISCUSSION THREAD

**Discussion by yacineblr** [evertramos/nginx-proxy-automation#300] [Q&A]
Context: I have 2 subdomains. 
- a.domain.com, www.a.domain.com
- b.domain.com

Hi, I have a problem with the SSL certificate. I have created a wordpress container (from your project) with a domain name (ex: A.domain.com), everything works fine. I modify the hosts in .env file (in the wordpress project), so that it points to a new domain (B.domain.com).
I think I need to completely recreate the containers so that the changes in the .env file are taken into account, I perform these actions in wordpress-project:

```
docker-compose stop 
docker-compose rm
docker-compose create 
docker-compose start
```

The site is accessible, from http://B.domain.com but the https causes an error because the certificate is not valid. I realize that I am missing the "www.B.domain.com", so I create the subdomain but the site is still not functional in https.

The questions are: 
- When are the SSL certificates generated?
         If I add a new wordpress site while the server is online, I just have to mount the docker-compose,
         and the new site will be available automatically? 
- How to force the regeneration of the SSL certificate?

**Comment by evertramos** [comment]
Hello @yacineblr!

When changing the url you must stop the container and wait a few seconds before starting again, in order to remove the nginx configuration from the proxy, then you start the container again. Please check the log from the letsencrypt container to see if there is any error on the certificate issuance.

**Reply by yacineblr** [reply]
Thank you for your answer. 
So to change the url I only have to change the .env file, restart the container and wait a few minutes?
What about wordpress? The url will be automatically changed (even for links) or I have to go through "wpcli"?

Sorry, I don't know much about infra and even less about wordpress. But your project will help me a lot, thanks.

**Comment by yacineblr** [comment]
When I execute the command to update the url of wordpress :
```
$ ./wp.sh option update home 'http://mydomain.ovh.com'
```
I get this error.

```
Creating mysite_wpcli_run ... done
Error: Error establishing a database connection.
ERROR: 1
```

**Reply by evertramos** [reply]
How would this be related to the proxy?

**Reply by yacineblr** [reply]
My bad, it's not related
